File: python/sort/bubble_sort.py
def bubble_sort1(array):
    if not isinstance(array, list):
        return False
    for i in reversed(xrange(1, len(array))):
        for j in xrange(0, i):
            if array[j] > array[j + 1]:
                array[j], array[j + 1] = array[j + 1], array[j]


def bubble_sort(array):
    if not isinstance(array, list):
        return False
    i = len(array) - 1
    while i > 0:
        for j in xrange(0, i):
            if array[j] > array[i]:
                array[j], array[i] = array[i], array[j]
        i -= 1

File: c/sort/bubble_sort.c
#include "sorts.h"

void bubble_sort(int array[], int length) {
    for (int i = length - 1; i > 0; i --) {
        for (int j  = 0; j < i - 1; j ++) {
            if (array[j] > array[j+1]) {
                array[j] = array[j] ^ array[j+1];
                array[j+1] = array[j] ^ array[j+1];
                array[j] = array[j] ^ array[j+1];
            }
        }
    }
}

